services/accounts/accounts.go

```diff
@@ -441,10 +453,12 @@ func (api *API) AddMigratedKeyPair(ctx context.Context, kcUID string, kpName str
 	}
 
 	// Once we migrate a keypair, corresponding keystore files need to be deleted.
-	for _, addr := range addresses {
-		// This action deletes an account from the keystore, no need to check for error in this context here, cause if this
-		// action fails from whichever reason the account is still successfully migrated since keystore won't be used any more.
-		_ = api.manager.DeleteAccount(addr, "")
+	if len(password) > 0 {
+		for _, addr := range addresses {
+			// This action deletes an account from the keystore, no need to check for error in this context here, cause if this
+			// action fails from whichever reason the account is still successfully migrated since keystore won't be used any more.
```

Comment:
I know that this comment is not added specifically in this PR, but I still don't fully understand why it is safe to ignore any error here. Considering we want to remove file, and we do not because of some error, it might a problem from security pov to not handle this error (or at least to not notify user about it) even if everything was migrated as expected. Am I missing something?

Reply:
@rasom yes, I agree. That originates from the old `DeleteAccount` function which was not removing cache properly and I was thinking in that time about the case if somebody manually removes the keystore from the file system that we could just simply ignore that and complete the action with success. But I agree, will change that.